src/segment_regquery.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+type regquery struct {
+	props     *properties
+	env       environmentInfo
+	content   string
+	errorInfo string
+}
+
+const (
+
+	// Properties defining how to get to the key.
+	RegistryPath Property = "registry_path" // path from the supplied root under which the key exists
+	RegistryKey  Property = "registry_key"  // key within full reg path formed from two above
+
+	// Properties defining fallback behaviour in case the query fails.
+	QueryFailBehaviour      Property = "query_fail_behaviour"       // what to do, can be one of "hide_segment", "display_fallback_string", "show_debug_info"
+	QueryFailFallbackString Property = "query_fail_fallback_string" // what to display if above is "display_fallback_string"
```

Comment:
I would drop this and hide the segment on error or no value. If we move the actual implementation code to `environment_windows.go`, we can use oh-my-posh's native debug features to let people know what happens in case the segment isn't shown.

Reply:
the fallback string method is useful - in a particular case for which I was writing this, the lack of existence the key in the registry defines a default state that it's useful to display.

totally agree on the debug though, much better to have that logged via the env's debug.

Reply:
I agree, but the I'd log that straight in the debug log. But I think that's also what you meant.

Reply:
Actually, the fallback string is intended for the segment, not debug.  Use case is that I am querying the registry for what a certain toolchain puts in there based on how I configure it.  When there is a value there, it defines the debug environment I am in, when it's not there it will be using the live environment.  As a cmd prompt user, I want to be able to clearly see both scenarios (non-existence of the segment is not explicit enough).

There are certainly other cases where the segment could just not show.  I'd be happy to share some more details on specifics privately.

Reply:
Aha - looks like this could be cleaner done via template system, grokking how that works now.

Reply:
done via templates